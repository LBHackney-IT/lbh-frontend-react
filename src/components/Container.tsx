import classNames from "classnames";
import PropTypes from "prop-types";
import React from "react";

import { Attributes, DataAttributes } from "../helpers/Attributes";

import "lbh-frontend/lbh/objects/_width-container.scss";

/**
 * The proptypes for {@link Container}.
 *
 * This also supports all `aria-*` and `data-*` attributes.
 *
 * @noInheritDoc
 */
export interface ContainerProps extends React.AriaAttributes, DataAttributes {
  id?: string;
  className?: string;
  /**
   * The elements inside {@link Container}.
   */
  children: React.ReactNode;
}

/**
 * A component that wraps a set of elements within a single container. Can be
 * used to provide consistency in layout, for example, for all elements on a
 * page to have a certain width.
 */
export const Container: React.FunctionComponent<ContainerProps> = (
  props: ContainerProps
): JSX.Element => {
  const { id, className, children } = props;

  const extraAttributes = Attributes.ariaAndData(props);

  return (
    <div
      id={id}
      className={classNames("govuk-container", "lbh-container", className)}
      {...extraAttributes}
    >
      {children}
    </div>
  );
};

Container.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};
