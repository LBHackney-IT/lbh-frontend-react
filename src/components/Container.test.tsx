import React from "react";
import { create } from "react-test-renderer";

import { Container } from "./Container";

it("renders correctly with all props", () => {
  const component = create(
    <Container
      id="test"
      className="class1 class2"
      aria-label="test container"
      data-test="test data"
    >
      Test
    </Container>
  );

  expect(component).toMatchInlineSnapshot(`
    <div
      aria-label="test container"
      className="govuk-container lbh-container class1 class2"
      data-test="test data"
      id="test"
    >
      Test
    </div>
  `);
});

it("renders correctly without optional props", () => {
  const component = create(<Container>Test</Container>);

  expect(component).toMatchInlineSnapshot(`
    <div
      className="govuk-container lbh-container"
    >
      Test
    </div>
  `);
});
