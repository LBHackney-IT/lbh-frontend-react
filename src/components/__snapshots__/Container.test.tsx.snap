// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<span
  className="govuk-container lbh-container class1 class2"
  id="test"
>
  Test
</span>
`;

exports[`renders correctly without optional props 1`] = `
<span
  className="govuk-container lbh-container"
>
  Test
</span>
`;
